src/EnergyPlus/ThermalComfort.cc

```diff
@@ -2545,25 +2541,30 @@ namespace ThermalComfort {
             state.dataThermalComforts->ThermalComfortSetPoint(iZone).notMetHeating = 0.0;
             state.dataThermalComforts->ThermalComfortSetPoint(iZone).notMetCoolingOccupied = 0.0;
             state.dataThermalComforts->ThermalComfortSetPoint(iZone).notMetHeatingOccupied = 0.0;
-            {
-                auto const SELECT_CASE_var(state.dataHeatBalFanSys->TempControlType(iZone));
-                if (SELECT_CASE_var == SingleHeatingSetPoint) {
-                    testHeating = true;
-                    testCooling = false;
-                } else if (SELECT_CASE_var == SingleCoolingSetPoint) {
-                    testHeating = false;
-                    testCooling = true;
-                } else if (SELECT_CASE_var == SingleHeatCoolSetPoint) {
-                    testHeating = true;
-                    testCooling = true;
-                } else if (SELECT_CASE_var == DualSetPointWithDeadBand) {
-                    testHeating = true;
-                    testCooling = true;
-                } else {
-                    testHeating = true;
-                    testCooling = true;
-                }
+
+            switch (state.dataHeatBalFanSys->TempControlType(iZone)) {
```

Comment:
Ummmmm ... same goes for using `switch/case` to set variables to `true`/`false`.